File: lang-java/dream/No2789.java
/**
 * @author <a href="https://github.com/Matthew-Han">Matthew Han</a>
 * @since 2024/3/14 15:35:46
 **/
public class No2789 {

    /**
     * 合并后数组中的最大元素
     *
     * @param nums
     * @return
     */
    public long maxArrayValue(int[] nums) {
        long ans = nums[0];
        long tmp = nums[nums.length - 1];
        for (int r = nums.length - 1; r > 0; r--) {
            tmp = nums[r - 1] <= tmp ? tmp + nums[r - 1] : nums[r - 1];
            ans = Math.max(ans, tmp);
        }
        return ans;
    }
}

File: lang-py/No2789.py
"""
  author <a href="https://github.com/Matthew-Han">Matthew Han</a>
  since 2024/3/14 15:35:30
"""
from typing import List


class Solution:
    def maxArrayValue(self, nums: List[int]) -> int:
        ans = nums[0]
        tmp = nums[-1]
        for i in range(len(nums), 0, -1):
            tmp = nums[i - 1] + tmp if nums[i - 1] <= tmp else nums[i - 1]
            ans = max(ans, tmp)
        return ans
